astropy/io/fits/util.py

```diff
@@ -882,3 +882,23 @@ def _free_space_check(hdulist, dirname=None):
             hdu._close()
 
         raise OSError(error_message + str(exc))
+
+
+def _extract_number(value, default):
+    """
+    Attempts to extract an integer number from the given value. If the
+    extraction fails, the value of the 'default' argument is returned.
+    """
+
+    try:
```

Comment:
Well, there is a subtle difference - the `_str_to_num` in its current state will return floats for strings like "9.0" -> 9.0. It will also raise an exception for any value which has unsupported type.

The `_extract_number` is more robust in that manner, and the name does not explicitly suggest that it deals with strings as inputs. Hence if we want to avoid having two similiar functions, I would prefer that `_extract_number` replaces `_str_to_num`, but making that decision requires some investigation.